File: 242_ValidAnagram.java
// Given two strings s and t, write a function to determine if t is an anagram of s.
// For example,
// s = "anagram", t = "nagaram", return true.
// s = "rat", t = "car", return false.

//Idea: sort two String usign java.util.Arrays.sort(), compare the sorted Strings. The time complexity is O(nlogn)

public class Solution {
    public boolean isAnagram(String s, String t) {
        char[] c1 = s.toCharArray();
        char[] c2 = t.toCharArray();
        
        Arrays.sort(c1);
        Arrays.sort(c2);
        
        String s1 = String.valueOf(c1);
        String s2 = String.valueOf(c2);
        
        return s1.equals(s2);
    }
}

File: PYTHON/200-299/242_Valid_Anagram.py
#Given two strings s and t, write a function to determine if t is an anagram of s. 
#
#For example,
#s = "anagram", t = "nagaram", return true.
#s = "rat", t = "car", return false.
#
#
#Note:
#You may assume the string contains only lowercase alphabets.
#
#Follow up:
#What if the inputs contain unicode characters? How would you adapt your solution to such case?

class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        dict = {}
        for i in s:
            if not dict.has_key(i):
                dict[i] = 1
            else:
                dict[i] += 1
        
        for i in t:
            if not dict.has_key(i):
                return False
            else:
                dict[i] -= 1
        
        for i in dict:
            if dict[i] != 0:
                return False
        
        return True